src/Intersection.ts

```diff
@@ -81,6 +81,36 @@ export class Intersection {
     }
   }
 
+  /**
+   * Use the ray casting algorithm to determine if {@link point} is in the polygon defined by {@link points}
+   * @see https://en.wikipedia.org/wiki/Point_in_polygon
+   * @param point
+   * @param points polygon points
+   * @returns
+   */
+  static isPointInPolygon(point: Point, points: Point[]) {
+    const other = new Point(point).setX(
+      Math.min(point.x - 1, ...points.map((p) => p.x))
+    );
```

Comment:
this is a simple workaround to the need to cast a ray - I could have refactored intersectsLineLine for that but decided not to, keeping things simple

Reply:
Is the scope of this loop to define a point on the left that is for sure outside the polygon?

Reply:
If the answer is yes, i would rather use -Number.MAX_VALUE and ditch the loop.

Reply:
yes, instead of casting an infinite ray.
but you are right. we must reduce it by another step

Reply:
I have tried to change the intersection methods to support ray casting. Not a lot of change. and I think it is better than this

Reply:
Half of the speed hit goes away, just to say.
map on 4 points means creating 4 function stacks, creating a new array, then spread it, and run Math.min on it. on those things that do pure calculations is all wasted time.